### DISCUSSION TITLE
Classify links into categories

### DISCUSSION THREAD

**Discussion by cedriclocqueneux** [YOURLS/YOURLS#2956] [Ideas]
Hello,

I am surprised that there is no possibility to sort the links into folders or categories. 
For example, most of my shortcut links point to my site and are used on social networks. But right now I am creating shortcut links that will be used in a paper book. And I would have liked to be able to classify these links in a specific category so that I could find them easily, with one click, to see the stats of the links used only in this book.
Unless you have another solution? I did think of "coding" for example the links with a prefix, like yourls.com/bk-link1, yourls.com/bk-link2, etc. But it will still be inconvenient to consult in the admin

**Comment by ozh** [comment]
Yeah, long time feature request, see for instance #1922. Might come some day. Will come some day ! :)

**Reply by cedriclocqueneux** [reply]
I hope! Because it becomes quickly complicated to find your way among hundreds, even thousands, of links :( Yourls is excellent, light and easy to use, and can be completed with some nice plugins. The only thing missing is the possibility to classify the links, either in categories or tags.

**Comment by thisiszeev** [comment]
What is needed to make this happen?

**Reply by dgw** [reply]
@LeoColomb Did you build the v2 rewrite in such a way that `Link` categories would be easy to add?

I don't believe this is particularly easy to do in the current release of YOURLS, but planning ahead is always good. A rewrite has been in process for some time now.

**Reply by LeoColomb** [reply]
@dgw Yes, it would. Or more precisely, any attributes/labels, then categories are just another way to call them.

**Reply by thisiszeev** [reply]
Well I am interested in helping with this. I am just busy at work currently preparing for a massive relaunch. But by April I should have some time free again, hopefully.

**Comment by thisiszeev** [comment]
I was thinking, creating tags for shortened urls would be better than folders/categories. This would allow for multiple tags per url, and easy filtering. Would also mean no changes to the existing database scheme, just adding a table.

**Reply by LeoColomb** [reply]
Yup. Attributes, labels, and tags are three words meaning the same in this situation. So it's the goal 😃

**Reply by pascalinger** [reply]
Tags sound like a great idea. We any close to achieving this?
